zookeeper-contrib/zookeeper-contrib-zabtla/experiment/ZabWithQTest.tla

```diff
@@ -0,0 +1,994 @@
+(*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *)
+-------------------------------- MODULE ZabWithQTest --------------------------------
+\* This is the test for formal specification for the Zab consensus algorithm,
```

Comment:
Can we make this module extends Zab module to avoid copy / paste?

Reply:
Yes, we can. We need to ensure that Zab.tla and ZabWithQTest.tla are in the same directory in this situation. So here we can put these two files in the same directory or copy Zab.tla into experiment/. I think what you suggest will make the experimental framework look clearer. Thank you!

Reply:
Thinking more about the folder structure of the  specification files, I am tempted to make a proposal that instead of putting them under zookeeper-contrib, we create a folder under root level called zookeeper-specifications, and put ZAB specifications there. I feel these specifications have more affinities to the zookeeper core project comparing to zookeeper-contrib. 

The new root folder structures would look like:
├── bin
├── conf
├── dev
├── target
├── tools
├── zookeeper-assembly
├── zookeeper-client
├── zookeeper-compatibility-tests
├── zookeeper-contrib
├── zookeeper-docs
├── zookeeper-it
├── zookeeper-jute
├── zookeeper-metrics-providers
├── zookeeper-recipes
├── zookeeper-server
└── zookeeper-specifications

And the spec folder looks like
└── zookeeper-specifications
    ├── leader-election
    ├── reconfig
    └── zab

Here we put the zab spec, toolbox configs, tests, etc in zab folder.